gsa/src/web/pages/tickets/component.js

```diff
@@ -0,0 +1,281 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React from 'react';
+
+import {connect} from 'react-redux';
+
+import {_} from 'gmp/locale/lang';
+
+import {ALL_FILTER} from 'gmp/models/filter';
+
+import {selectSaveId} from 'gmp/utils/id';
+import {isDefined} from 'gmp/utils/identity';
+
+import EntityComponent from 'web/entity/component';
+
+import {
+  loadEntities as loadUsers,
+  selector as usersSelector,
+} from 'web/store/entities/users';
+
+import compose from 'web/utils/compose';
+import PropTypes from 'web/utils/proptypes';
+import withGmp from 'web/utils/withGmp';
+
+import CreateTicketDialog from './createdialog';
+import StatusChangeTicketDialog from './statuschangedialog';
+
+class TicketComponent extends React.Component {
+
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      createDialogVisible: false,
+      solvedDialogVisible: false,
+    };
+
+    this.handleCloseCreateDialog = this.handleCloseCreateDialog.bind(this);
+    this.handleOpenCreateDialog = this.handleOpenCreateDialog.bind(this);
+    this.handleCloseSolvedDialog = this.handleCloseSolvedDialog.bind(this);
+    this.handleOpenSolvedDialog = this.handleOpenSolvedDialog.bind(this);
+    this.handleCloseClosedDialog = this.handleCloseClosedDialog.bind(this);
+    this.handleOpenClosedDialog = this.handleOpenClosedDialog.bind(this);
+
+    this.handleSolve = this.handleSolve.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  handleOpenCreateDialog(result) {
+    this.props.loadUsers();
+
+    this.setState({
+      result,
+      createDialogVisible: true,
+    });
+
+    this.handleInteraction();
+  }
+
+  handleCloseCreateDialog() {
+    this.setState({createDialogVisible: false});
+
+    this.handleInteraction();
+  }
+
+  handleOpenSolvedDialog(ticket) {
+    this.setState({
+      ticket,
+      solvedDialogVisible: true,
+    });
+
+    this.handleInteraction();
+  }
+
+  handleCloseSolvedDialog() {
+    this.setState({solvedDialogVisible: false});
+
+    this.handleInteraction();
+  }
+
+  handleOpenClosedDialog(ticket) {
+    this.setState({
+      ticket,
+      closedDialogVisible: true,
+    });
+
+    this.handleInteraction();
+  }
+
+  handleCloseClosedDialog() {
+    this.setState({closedDialogVisible: false});
+
+    this.handleInteraction();
+  }
+
+  handleUserIdChange(userId) {
+    this.setState({userId});
+  }
+
+  handleInteraction() {
+    const {onInteraction} = this.props;
+    if (isDefined(onInteraction)) {
+      onInteraction();
+    }
+  }
+
+  handleSolve({
+    ticketId,
+    comment,
+  }) {
+    const {
+      gmp,
+      onSolved,
+      onSolveError,
+    } = this.props;
+
+    this.setState({solvedDialogVisible: false});
```

Comment:
Should the dialog close before the error handling is done? I think it would be nice to have a possible error message in the red box within the dialog. Same for handleClose()